File: 1710_Maximum_Units_on_a_Truck.java
import java.util.*;

class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, Comparator.comparingDouble(o -> o[1]));
        int spaceRemaining = truckSize;
        int sum = 0;
        for (int i = boxTypes.length - 1; i >= 0; i--) {
            if (spaceRemaining >= boxTypes[i][0]) {
                sum += boxTypes[i][0] * boxTypes[i][1];
                spaceRemaining -= boxTypes[i][0];
            } else {
                sum += spaceRemaining * boxTypes[i][1];
                spaceRemaining = 0;
            }
            if (spaceRemaining == 0) break;
        }
        return sum;
    }
}

File: 1710_Maximum_Units_on_a_Truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxes = []
        for box in boxTypes:
            boxes.append({
                "quantity": box[0],
                "units": box[1]
            })
            
        boxes.sort(key = itemgetter("units"), reverse = True)
        current_box = 0
        units_loaded = 0
        while truckSize > 0 and current_box < len(boxes):
            while boxes[current_box]["quantity"] > 0 and truckSize > 0:
                truckSize -= 1
                units_loaded += boxes[current_box]["units"]
                boxes[current_box]["quantity"] -= 1
            current_box += 1
        return units_loaded
            